### ISSUE TITLE
Failing test during build

### ISSUE THREAD

**Issue by GeoffreyOnRails** [nilmtk/nilmtk#526]
I've followed the installations steps for Ubuntu using Anaconda, but when running the tests, I have a failed test :

(nilmtk-env) gvo@gvo:~/nilmtk$ nosetests
........F..................................
======================================================================
FAIL: test_load (nilmtk.tests.test_datastore.TestCSVDataStore)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/gvo/nilmtk/nilmtk/tests/test_datastore.py", line 40, in test_load
    self.assertEqual(df.look_ahead.index[0], timeframe.end)
AssertionError: Timestamp('2012-01-01 00:00:06') != Timestamp('2012-01-01 00:00:05')

----------------------------------------------------------------------
Ran 43 tests in 25.413s

FAILED (failures=1)
Closing remaining open files:/home/gvo/nilmtk/data/energy.h5...done

I have been able to reproduce it on the CI environnement : https://travis-ci.org/GeoffreyOnRails/nilmtk/builds/165311704, so I suspect it is related to a new version of a dependency that might behave differently.

**Comment by GeoffreyOnRails** [comment]
After some investigation, surprisingly, the tests are OK when running only the correct file. (nosetests nilmtk/tests/test_datastore.py -v -s)
df.lookahead ->                          1      2      3
0                                       
2012-01-01 00:00:05  391.0  423.0  773.0
2012-01-01 00:00:06  846.0  531.0   80.0
2012-01-01 00:00:07  607.0  186.0  133.0
2012-01-01 00:00:08  166.0  987.0  649.0
2012-01-01 00:00:09  136.0  740.0  161.0
2012-01-01 00:00:10  716.0  822.0  842.0
2012-01-01 00:00:11  586.0  861.0  987.0
2012-01-01 00:00:12  214.0  700.0  314.0
2012-01-01 00:00:13  718.0  877.0  116.0
2012-01-01 00:00:14  778.0  524.0  317.0

While when I run all the tests : 

df.lookahead ->                           1      2      3
0                                       
2012-01-01 00:00:06  846.0  531.0   80.0
2012-01-01 00:00:07  607.0  186.0  133.0
2012-01-01 00:00:08  166.0  987.0  649.0
2012-01-01 00:00:09  136.0  740.0  161.0
2012-01-01 00:00:10  716.0  822.0  842.0
2012-01-01 00:00:11  586.0  861.0  987.0
2012-01-01 00:00:12  214.0  700.0  314.0
2012-01-01 00:00:13  718.0  877.0  116.0
2012-01-01 00:00:14  778.0  524.0  317.0
2012-01-01 00:00:15  987.0  881.0   70.0

Thus leading to the error.

**Comment by nipunbatra** [comment]
Is this still a problem for you? Cause, the tests run fine for me.

**Comment by GeoffreyOnRails** [comment]
Hello, 

Tests now run fine for me also now, but I suspect it is because the previously failing test have simply been commented with the following comment : 

>        # self.assertEqual(df.look_ahead.index[0], timeframe.end) # This test, for some odd reason, fails for CSVDataStore in Python3, intermittently.  Very odd.

This has been done in commit 9e63ea65dd1c74452ba68659c0339425e4f4c6b8, authored by Jack Kelly.

I suspect the comment is wrong, it indeed fails intermittently (see above) but I had the same issue without Python3, so it is likely not Python3-related.

In conclusion, the issue with failing tests is "temporarily fixed" only, there is still a test (and potentially a bug somewhere) that needs to be fixed. Therefore, I would rather not close this issue.

I'll try to investigate a little bit further if I manage to find some time.

**Comment by itroulli** [comment]
I'm failing the test too with these errors:
`2010-01-01 00:00:00       0.0
2010-01-01 00:00:10       0.0
2010-01-01 00:00:20       0.0
2010-01-01 00:00:30     100.0
2010-01-01 00:03:20     100.0
2010-01-01 00:03:30     100.0
2010-01-01 00:03:40     150.0
2010-01-01 00:03:50     150.0
2010-01-01 00:04:00     200.0
2010-01-01 00:04:09       0.0
2010-01-01 00:04:20       0.0
2010-01-01 00:04:30     100.0
2010-01-01 00:04:50    5000.0
2010-01-01 00:16:40       0.0
Name: (power, active), dtype: float32
Calculating dropout_rate for ElecMeterID(instance=1, building=1, dataset='REDD') ...   
--------------------- >> end captured stdout << ----------------------

======================================================================
ERROR: test_pipeline (nilmtk.stats.tests.test_locategoodsections.TestLocateGoodSections)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ilias/nilmtk/nilmtk/stats/tests/test_locategoodsections.py", line 51, in test_pipeline
    meter.good_sections(**kwargs)
  File "/home/ilias/nilmtk/nilmtk/elecmeter.py", line 625, in good_sections
    nodes, results_obj, loader_kwargs)
  File "/home/ilias/nilmtk/nilmtk/elecmeter.py", line 664, in _get_stat_from_cache_or_compute
    results_obj_copy = deepcopy(results_obj)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 190, in deepcopy
    y = _reconstruct(x, rv, 1, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 334, in _reconstruct
    state = deepcopy(state, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 163, in deepcopy
    y = copier(x, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 257, in _deepcopy_dict
    y[deepcopy(key, memo)] = deepcopy(value, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 190, in deepcopy
    y = _reconstruct(x, rv, 1, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 334, in _reconstruct
    state = deepcopy(state, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 163, in deepcopy
    y = copier(x, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 257, in _deepcopy_dict
    y[deepcopy(key, memo)] = deepcopy(value, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 190, in deepcopy
    y = _reconstruct(x, rv, 1, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 334, in _reconstruct
    state = deepcopy(state, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 163, in deepcopy
    y = copier(x, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 237, in _deepcopy_tuple
    y.append(deepcopy(a, memo))
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 163, in deepcopy
    y = copier(x, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 230, in _deepcopy_list
    y.append(deepcopy(a, memo))
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 174, in deepcopy
    y = copier(memo)
ValueError: Iteration of zero-sized operands is not enabled

======================================================================
ERROR: test_co_correctness (nilmtk.tests.test_combinatorial_optimisation.TestCO)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ilias/nilmtk/nilmtk/tests/test_combinatorial_optimisation.py", line 29, in test_co_correctness
    co.disaggregate(mains, output, sample_period=1)
  File "/home/ilias/nilmtk/nilmtk/disaggregate/combinatorial_optimisation.py", line 148, in disaggregate
    load_kwargs.setdefault('sections', mains.good_sections())
  File "/home/ilias/nilmtk/nilmtk/elecmeter.py", line 625, in good_sections
    nodes, results_obj, loader_kwargs)
  File "/home/ilias/nilmtk/nilmtk/elecmeter.py", line 664, in _get_stat_from_cache_or_compute
    results_obj_copy = deepcopy(results_obj)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 190, in deepcopy
    y = _reconstruct(x, rv, 1, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 334, in _reconstruct
    state = deepcopy(state, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 163, in deepcopy
    y = copier(x, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 257, in _deepcopy_dict
    y[deepcopy(key, memo)] = deepcopy(value, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 190, in deepcopy
    y = _reconstruct(x, rv, 1, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 334, in _reconstruct
    state = deepcopy(state, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 163, in deepcopy
    y = copier(x, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 257, in _deepcopy_dict
    y[deepcopy(key, memo)] = deepcopy(value, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 190, in deepcopy
    y = _reconstruct(x, rv, 1, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 334, in _reconstruct
    state = deepcopy(state, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 163, in deepcopy
    y = copier(x, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 237, in _deepcopy_tuple
    y.append(deepcopy(a, memo))
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 163, in deepcopy
    y = copier(x, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 230, in _deepcopy_list
    y.append(deepcopy(a, memo))
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 174, in deepcopy
    y = copier(memo)
ValueError: Iteration of zero-sized operands is not enabled
-------------------- >> begin captured stdout << ---------------------
Training model for submeter 'ElecMeter(instance=2, building=1, dataset=None, appliances=[])'
Training model for submeter 'ElecMeter(instance=3, building=1, dataset=None, appliances=[])'
Done training!

--------------------- >> end captured stdout << ----------------------

======================================================================
ERROR: test_proportion_of_energy (nilmtk.tests.test_elecmeter.TestElecMeter)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ilias/nilmtk/nilmtk/tests/test_elecmeter.py", line 66, in test_proportion_of_energy
    self.assertEquals(meter.proportion_of_energy(meter), 1.0)
  File "/home/ilias/nilmtk/nilmtk/electric.py", line 284, in proportion_of_energy
    good_other_sections = other.good_sections(**loader_kwargs)
  File "/home/ilias/nilmtk/nilmtk/elecmeter.py", line 625, in good_sections
    nodes, results_obj, loader_kwargs)
  File "/home/ilias/nilmtk/nilmtk/elecmeter.py", line 664, in _get_stat_from_cache_or_compute
    results_obj_copy = deepcopy(results_obj)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 190, in deepcopy
    y = _reconstruct(x, rv, 1, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 334, in _reconstruct
    state = deepcopy(state, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 163, in deepcopy
    y = copier(x, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 257, in _deepcopy_dict
    y[deepcopy(key, memo)] = deepcopy(value, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 190, in deepcopy
    y = _reconstruct(x, rv, 1, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 334, in _reconstruct
    state = deepcopy(state, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 163, in deepcopy
    y = copier(x, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 257, in _deepcopy_dict
    y[deepcopy(key, memo)] = deepcopy(value, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 190, in deepcopy
    y = _reconstruct(x, rv, 1, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 334, in _reconstruct
    state = deepcopy(state, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 163, in deepcopy
    y = copier(x, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 237, in _deepcopy_tuple
    y.append(deepcopy(a, memo))
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 163, in deepcopy
    y = copier(x, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 230, in _deepcopy_list
    y.append(deepcopy(a, memo))
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 174, in deepcopy
    y = copier(memo)
ValueError: Iteration of zero-sized operands is not enabled

======================================================================
ERROR: test_total_energy (nilmtk.tests.test_elecmeter.TestElecMeter)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ilias/nilmtk/nilmtk/tests/test_elecmeter.py", line 38, in test_total_energy
    meter.total_energy()
  File "/home/ilias/nilmtk/nilmtk/elecmeter.py", line 586, in total_energy
    nodes, TotalEnergy.results_class(), loader_kwargs)
  File "/home/ilias/nilmtk/nilmtk/elecmeter.py", line 664, in _get_stat_from_cache_or_compute
    results_obj_copy = deepcopy(results_obj)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 190, in deepcopy
    y = _reconstruct(x, rv, 1, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 334, in _reconstruct
    state = deepcopy(state, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 163, in deepcopy
    y = copier(x, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 257, in _deepcopy_dict
    y[deepcopy(key, memo)] = deepcopy(value, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 190, in deepcopy
    y = _reconstruct(x, rv, 1, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 334, in _reconstruct
    state = deepcopy(state, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 163, in deepcopy
    y = copier(x, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 257, in _deepcopy_dict
    y[deepcopy(key, memo)] = deepcopy(value, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 190, in deepcopy
    y = _reconstruct(x, rv, 1, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 334, in _reconstruct
    state = deepcopy(state, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 163, in deepcopy
    y = copier(x, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 237, in _deepcopy_tuple
    y.append(deepcopy(a, memo))
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 163, in deepcopy
    y = copier(x, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 230, in _deepcopy_list
    y.append(deepcopy(a, memo))
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 174, in deepcopy
    y = copier(memo)
ValueError: Iteration of zero-sized operands is not enabled

======================================================================
ERROR: test_fhmm_correctness (nilmtk.tests.test_fhmm.TestFHMM)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ilias/nilmtk/nilmtk/tests/test_fhmm.py", line 28, in test_fhmm_correctness
    output = HDFDataStore('output.h5', 'w')
  File "/home/ilias/nilmtk/nilmtk/docinherit.py", line 46, in f
    return self.mthd(obj, *args, **kwargs)
  File "/home/ilias/nilmtk/nilmtk/datastore/hdfdatastore.py", line 22, in __init__
    self.store = pd.HDFStore(filename, mode, complevel=9, complib='blosc')
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/site-packages/pandas/io/pytables.py", line 448, in __init__
    self.open(mode=mode, **kwargs)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/site-packages/pandas/io/pytables.py", line 612, in open
    raise e
ValueError: The file 'output.h5' is already opened.  Please close it before reopening in write mode.
-------------------- >> begin captured stdout << ---------------------
Training model for submeter 'ElecMeter(instance=2, building=1, dataset=None, appliances=[])'
Training model for submeter 'ElecMeter(instance=3, building=1, dataset=None, appliances=[])'

--------------------- >> end captured stdout << ----------------------

======================================================================
ERROR: test_dual_supply (nilmtk.tests.test_metergroup.TestMeterGroup)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ilias/nilmtk/nilmtk/tests/test_metergroup.py", line 95, in test_dual_supply
    self.assertEqual(mg['washer dryer'].total_energy()['active'],
  File "/home/ilias/nilmtk/nilmtk/metergroup.py", line 829, in total_energy
    load_kwargs, 'total_energy', full_results)
  File "/home/ilias/nilmtk/nilmtk/metergroup.py", line 845, in _collect_stats_on_all_meters
    **load_kwargs)
  File "/home/ilias/nilmtk/nilmtk/elecmeter.py", line 586, in total_energy
    nodes, TotalEnergy.results_class(), loader_kwargs)
  File "/home/ilias/nilmtk/nilmtk/elecmeter.py", line 664, in _get_stat_from_cache_or_compute
    results_obj_copy = deepcopy(results_obj)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 190, in deepcopy
    y = _reconstruct(x, rv, 1, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 334, in _reconstruct
    state = deepcopy(state, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 163, in deepcopy
    y = copier(x, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 257, in _deepcopy_dict
    y[deepcopy(key, memo)] = deepcopy(value, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 190, in deepcopy
    y = _reconstruct(x, rv, 1, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 334, in _reconstruct
    state = deepcopy(state, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 163, in deepcopy
    y = copier(x, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 257, in _deepcopy_dict
    y[deepcopy(key, memo)] = deepcopy(value, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 190, in deepcopy
    y = _reconstruct(x, rv, 1, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 334, in _reconstruct
    state = deepcopy(state, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 163, in deepcopy
    y = copier(x, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 237, in _deepcopy_tuple
    y.append(deepcopy(a, memo))
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 163, in deepcopy
    y = copier(x, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 230, in _deepcopy_list
    y.append(deepcopy(a, memo))
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 174, in deepcopy
    y = copier(memo)
ValueError: Iteration of zero-sized operands is not enabled
-------------------- >> begin captured stdout << ---------------------
Calculating total_energy for ElecMeterID(instance=1, building=1, dataset='REDD') ...   
--------------------- >> end captured stdout << ----------------------

======================================================================
ERROR: test_proportion_of_energy_submetered (nilmtk.tests.test_metergroup.TestMeterGroup)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ilias/nilmtk/nilmtk/tests/test_metergroup.py", line 78, in test_proportion_of_energy_submetered
    self.assertEqual(mg.proportion_of_energy_submetered(), 1.0)
  File "/home/ilias/nilmtk/nilmtk/metergroup.py", line 1053, in proportion_of_energy_submetered
    prop = m.proportion_of_energy(mains, **loader_kwargs)
  File "/home/ilias/nilmtk/nilmtk/electric.py", line 284, in proportion_of_energy
    good_other_sections = other.good_sections(**loader_kwargs)
  File "/home/ilias/nilmtk/nilmtk/elecmeter.py", line 625, in good_sections
    nodes, results_obj, loader_kwargs)
  File "/home/ilias/nilmtk/nilmtk/elecmeter.py", line 664, in _get_stat_from_cache_or_compute
    results_obj_copy = deepcopy(results_obj)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 190, in deepcopy
    y = _reconstruct(x, rv, 1, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 334, in _reconstruct
    state = deepcopy(state, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 163, in deepcopy
    y = copier(x, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 257, in _deepcopy_dict
    y[deepcopy(key, memo)] = deepcopy(value, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 190, in deepcopy
    y = _reconstruct(x, rv, 1, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 334, in _reconstruct
    state = deepcopy(state, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 163, in deepcopy
    y = copier(x, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 257, in _deepcopy_dict
    y[deepcopy(key, memo)] = deepcopy(value, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 190, in deepcopy
    y = _reconstruct(x, rv, 1, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 334, in _reconstruct
    state = deepcopy(state, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 163, in deepcopy
    y = copier(x, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 237, in _deepcopy_tuple
    y.append(deepcopy(a, memo))
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 163, in deepcopy
    y = copier(x, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 230, in _deepcopy_list
    y.append(deepcopy(a, memo))
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 174, in deepcopy
    y = copier(memo)
ValueError: Iteration of zero-sized operands is not enabled
-------------------- >> begin captured stdout << ---------------------
Running MeterGroup.proportion_of_energy_submetered...

--------------------- >> end captured stdout << ----------------------

======================================================================
ERROR: test_total_energy (nilmtk.tests.test_metergroup.TestMeterGroup)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ilias/nilmtk/nilmtk/tests/test_metergroup.py", line 132, in test_total_energy
    ds.buildings[1].elec.total_energy()
  File "/home/ilias/nilmtk/nilmtk/metergroup.py", line 829, in total_energy
    load_kwargs, 'total_energy', full_results)
  File "/home/ilias/nilmtk/nilmtk/metergroup.py", line 845, in _collect_stats_on_all_meters
    **load_kwargs)
  File "/home/ilias/nilmtk/nilmtk/elecmeter.py", line 586, in total_energy
    nodes, TotalEnergy.results_class(), loader_kwargs)
  File "/home/ilias/nilmtk/nilmtk/elecmeter.py", line 664, in _get_stat_from_cache_or_compute
    results_obj_copy = deepcopy(results_obj)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 190, in deepcopy
    y = _reconstruct(x, rv, 1, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 334, in _reconstruct
    state = deepcopy(state, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 163, in deepcopy
    y = copier(x, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 257, in _deepcopy_dict
    y[deepcopy(key, memo)] = deepcopy(value, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 190, in deepcopy
    y = _reconstruct(x, rv, 1, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 334, in _reconstruct
    state = deepcopy(state, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 163, in deepcopy
    y = copier(x, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 257, in _deepcopy_dict
    y[deepcopy(key, memo)] = deepcopy(value, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 190, in deepcopy
    y = _reconstruct(x, rv, 1, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 334, in _reconstruct
    state = deepcopy(state, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 163, in deepcopy
    y = copier(x, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 237, in _deepcopy_tuple
    y.append(deepcopy(a, memo))
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 163, in deepcopy
    y = copier(x, memo)
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 230, in _deepcopy_list
    y.append(deepcopy(a, memo))
  File "/home/ilias/anaconda2/envs/nilmtk-env/lib/python2.7/copy.py", line 174, in deepcopy
    y = copier(memo)
ValueError: Iteration of zero-sized operands is not enabled
-------------------- >> begin captured stdout << ---------------------
Calculating total_energy for ElecMeterID(instance=1, building=1, dataset=None) ...   
--------------------- >> end captured stdout << ----------------------

----------------------------------------------------------------------
Ran 43 tests in 16.505s

FAILED (errors=9)
Closing remaining open files:/home/ilias/nilmtk/data/random.h5...doneoutput.h5...done/home/ilias/nilmtk/data/energy.h5...done/home/ilias/nilmtk/data/co_test.h5...done
`
